module3/Projects/PairProject.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: page
+title: Mod3 Pair Project
+---
+
+# Coffee Shop
+
+## Kick Off
+
+[Learning to Pair](https://docs.google.com/presentation/d/1iSdA31sgik4ejeB95R4t2kvmh1p0rANC7VAfjc2Wgmo/edit#slide=id.gc6f90357f_0_0)
+
+### Objectives
+* Practice what we have learned about MVC applications
+* Work with a partner
+    * DTR
+    * Collaborative Git
+
+This project will help you practice the MVC skills we have learned, and push you to grow your collaborative skills. Each section of the project builds on the previous section; so, you will need to [pair program](https://www.geeksforgeeks.org/pair-programming/) with your partner.
+
+Before starting the project, read through the requirements (ask an instructor if you need any clarification).  Break down a list of tasks that need to be accomplished for each section, and determine who will 'drive' and who will 'navigate' for that task. Try to have equal opportunities to drive and navigate.
```

Comment:
That paragraph here makes it sound like we want them to do pair programming (two people working on same code at same time, driver/navigator style) for the whole project. I'm hesitant about this because like the slide deck says pair programming can be tiring, especially for introverted folks. But this project will also be a bit tricky to do with branching and git workflows because a lot depends on each other. 

Because this is the first time they are really doing git for pairs, my proposal is that up through CRUD for an Item, we are very prescriptive of what is pair programming and what is git branches. 

So pair programming through Create your MVC Application, with one person driving. Still pair programming through completing this user story, with the second partner driving because they cloned down the repo.

```
As a User
When I visit "/items"
Then I see a list of all items in the database
```

Then separate feature branches for details, new, edit, and delete. We should see 4 pull requests. 

Then for the Customer section, they get to choose how they want to structure the work. 

Thoughts?

Reply:
I like it!